ugvc/pipelines/lpr/filter_vcf_with_lib_prep_recalibration_model.py

```diff
@@ -0,0 +1,87 @@
+import os
+
+from simppl.cli import get_parser, get_simple_pipeline
+
+from ugvc.pipelines.mrd import featuremap_to_dataframe
+
+
+def init_parser():
+    parser = get_parser("filter_vcf_with_lib_prep_recalibration_model", run.__doc__)
+    parser.add_argument("--out_dir", required=True)
+    parser.add_argument("--output_suffix", required=True)
+    parser.add_argument("--ref_fasta", required=True)
+    parser.add_argument("--lib_prep_model_file", required=True)
+    parser.add_argument("--notebooks_dir", required=True)
+    parser.add_argument("--filtered_calls_vcf", help="filtered variant calling input vcf", required=True)
+    parser.add_argument("--truth_vcf", help='set of "true" calls vcf', required=True)
+    parser.add_argument("--featuremap_on_calls_vcf", help="featuremap vcf intersected on calls", required=True)
+    return parser
+
+
+def run(argv):
+    """
+    Filter vcf file using lib-prep recalibration model training pipeline
+    """
+    parser = init_parser()
+    sp = get_simple_pipeline(parser, argv=argv, name=__name__)
+    args = parser.parse_args(argv[1:])
+    print(args)
+
+    out_dir = args.out_dir
+    out_suffix = args.output_suffix
+    os.makedirs(out_dir, exist_ok=True)
+
+    filtered_calls_vcf = args.filtered_calls_vcf
+    truth_vcf = args.truth_vcf
+    featuremap_on_calls_vcf = args.featuremap_on_calls_vcf
+
+    calls_parquet = f"{out_dir}/calls_vcf.parquet"
+    featuremap_on_calls_parquet = f"{out_dir}/featuremap_on_calls.parquet"
+
+    sp.print_and_run_clt(
+        featuremap_to_dataframe.run,
+        [],
+        {
+            "-i": filtered_calls_vcf,
+            "-r": args.ref_fasta,
+            "-f": "TGCA",
+            "--info_fields_override": "",
+            "--format-fields": "DP=int AD=str SB=str VAF=str PL=str BG_AD=str",
+            "-o": calls_parquet,
+        },
+        flags=["--report-bases-in-reference-direction"],
+    )
+
+    sp.print_and_run_clt(
+        featuremap_to_dataframe.run,
+        [],
+        {"-i": featuremap_on_calls_vcf, "-r": args.ref_fasta, "-f": "TGCA", "-o": featuremap_on_calls_parquet},
+    )
+
+    tp_positions = f"{out_dir}/true_positions.csv"
+    sp.print_and_run(f"bcftools view {truth_vcf} -H -f PASS | " "awk '$6 > 10 {print $2}' > " + tp_positions)
```

Comment:
You can add "-i 'QUAL>10'" to bcftools instead of using awk, seems healthier to me.

Also, are you excluding 10 on purpose?

And maybe maybe it a parameter instead of hard coded?

Reply:
I was excluding 10 on purpose from the fresh-frozen.
Will externalize as a parameter.